### DISCUSSION TITLE
Regenerate citekeys on new computer

### DISCUSSION THREAD

**Discussion by thoresson** [retorquere/zotero-better-bibtex#2209] [Q&A] [answered]
Hi!

For the first time since I started to use Zotero, I'm on a new computer. I've synced my library using Zotero's sync service, but I run into problem with BBT's citekeys.

To my understanding, the citekeys generated by BBT on my old computer can't be synced to the new. So what I'm trying to is using the same template for my citekeys (auth.lower + year). This results in citekeys like _thoresson2022_, and when there are many citekeys for the same year and author, _thoresson2022a_ etc. 

But when generated on my new computer, the extra letter creating unique citekeys isn't added to the same source. So what is _thoresson2022a_ on my old computer and my new one are two different articles by Thoresson, both published in 2022.

Is this expected when regenerating citekeys on the news computer? Any suggestion on how to bring my old citekeys to the new one?

**Comment by retorquere** [comment, marked as answer]
Two ways to get them to the new computer:

1. The keys can be synced up, but you have to pin them all. That will sync over cleanly. In an upcoming release of Zotero (don't know when though), zotero items get a real citekey field, and then nothing else needs to be done.

2. If the old computer is fully synced up, you could just copy your Zotero profile (which includes the databases) to the new computer, and everything will be as it was. No need to pin in this case.

Those are unfortunately the only ways to get what you want. The behavior you see is indeed expected, but only because I know how BBT works in detail, I don't think anyone would expect this behavior. I have tried to get around this with ways of smuggling data across during sync, but none of them ended well; I have once had to abandon a Zotero account because of how thoroughly I screwed the account up, which is why I'm `emilianoeheyns` rather than `emilianoheyns` on the forum.

**Reply by thoresson** [reply]
Thanks!

**Reply by thoresson** [reply]
Forgot one thing yesterday: I'm also looking into using Zotero from both my personal and work computers. Is there any practical way to do that today, with BBT citekeys being synced? Pinning them all for now?

**Reply by retorquere** [reply]
Pinning them all is the only way. You might want to turn on [autoPinDelay](https://retorque.re/zotero-better-bibtex/installation/preferences/hidden-preferences/#autopindelay).

**Reply by thoresson** [reply]
Thanks again!

**Reply by thoresson** [reply]
Are there any downsides with pinning?

**Reply by retorquere** [reply]
Only if you expect substantial edits after creating the entry. If you set autopin eg to 1 minute, you have that minute after you import to do corrections. If you do edits after that you have to remember to refresh the key. BBT tries to keep the key the same even if you refresh (that is, it tries to keep the disambiguator letter the same if the base key remains the same).
